File: rocketmq-client/src/main/java/org/apache/rocketmq/client/route/Broker.java
package org.apache.rocketmq.client.route;

import javax.annotation.concurrent.Immutable;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import org.apache.rocketmq.client.remoting.Endpoints;

@AllArgsConstructor
@Getter
@EqualsAndHashCode
@Immutable
public class Broker {
    private final String name;
    private final int id;
    private final Endpoints endpoints;
}

File: csharp/rocketmq-client-csharp/Broker.cs
using rmq = Apache.Rocketmq.V2;

namespace Org.Apache.Rocketmq
{
    public class Broker
    {
        public Broker(rmq.Broker broker)
        {
            Name = broker.Name;
            Id = broker.Id;
            Endpoints = new Endpoints(broker.Endpoints);
        }

        public string Name { get; }
        public int Id { get; }
        public Endpoints Endpoints { get; }
    }
}